sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -1625,6 +1626,63 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("Expressions not supporting CS_AI collators") {
+    val unsupportedExpressions: Seq[Any] = Seq(
+      "ltrim",
+      "rtrim",
+      "trim",
+      "startswith",
+      "endswith",
+      "locate",
+      "instr",
+      "str_to_map",
+      "contains",
+      "replace",
+      ("translate", "efg"),
+      ("split_part", "2"),
+      ("substring_index", "2"))
+
+    val unsupportedCollator = "unicode_ai"
+    val supportedCollators: Seq[String] = Seq("unicode", "unicode_ci", "unicode_ci_ai")
+
+    unsupportedExpressions.foreach {
+      expression: Any =>
+        val unsupportedQuery: String = {
+          expression match {
+            case expression: String =>
+              s"select $expression('bcd' collate $unsupportedCollator, 'abc')"
+            case (expression: String, parameter: String) =>
+              s"select $expression('bcd' collate $unsupportedCollator, 'abc', '$parameter')"
+            case (expression: String, parameter: Integer) =>
+              s"select $expression('bcd' collate $unsupportedCollator, 'abc', $parameter)"
+          }
+        }
+
+        val analysisException = intercept[AnalysisException] {
+          sql(unsupportedQuery).collect()
+        }
+        assert(analysisException.getErrorClass === "DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE")
```

Comment:
use `checkError` (see examples elsewhere in the codebase) instead of `assert(analysisException.getErrorClass`